IM/ko_matrices.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+
+import numpy as np
+
+
+def get_konno_matrix(size: int, directory: Path):
+    """
+    Retrieves the precomputed Konno matrix from a file.
+
+    Parameters
+    ----------
+    size : int
+        The size of the matrix to load.
+    directory : Path
+        Directory containing precomputed KO matrices.
+
+    Returns
+    -------
+    np.ndarray
+        The loaded Konno matrix.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the required matrix file does not exist.
+    """
```

Comment:
```suggestion
def get_konno_matrix(size: int, directory: Path) -> np.memmap:
    """
    Retrieves the precomputed Konno matrix from a file.

    Parameters
    ----------
    size : int
        The size of the matrix to load.
    directory : Path
        Directory containing precomputed KO matrices.

    Returns
    -------
    np.memmap
        The loaded Konno matrix.

    Raises
    ------
    FileNotFoundError
        If the required matrix file does not exist.
    """
```

Just to emphasize that the array is memmap'd and not loaded into memory. There are also some subtle gotcha's with memmap vs regular numpy arrays and being aware of that in the type and docstring is useful.